File: Components/BlogML/Xml/BlogMLContent.vb
Imports System.Web
Imports System.Xml.Serialization

Namespace BlogML.Xml
 <Serializable()> _
 Public NotInheritable Class BlogMLContent
  Private ReadOnly Property Base64Encoded() As Boolean
   Get
    Return ContentType = ContentTypes.Base64
   End Get
  End Property

  Private ReadOnly Property HtmlEncoded() As Boolean
   Get
    Return ContentType = ContentTypes.Html OrElse ContentType = ContentTypes.Xhtml
   End Get
  End Property

  <XmlAttribute("type")> _
  Public Property ContentType() As ContentTypes
   Get
    Return m_ContentType
   End Get
   Set(value As ContentTypes)
    m_ContentType = Value
   End Set
  End Property
  Private m_ContentType As ContentTypes = ContentTypes.Text

  ' Encoded Text
  <XmlText()> _
  Public Property Text() As String
   Get
    Return m_Text
   End Get
   Set(value As String)
    m_Text = Value
   End Set
  End Property
  Private m_Text As String

  <XmlIgnore()> _
  Public ReadOnly Property UncodedText() As String
   Get
    If Base64Encoded Then
     Dim byteArray As Byte() = Convert.FromBase64String(Text)
     Return System.Text.Encoding.UTF8.GetString(byteArray)
    End If
    If HtmlEncoded Then
     Return HttpUtility.HtmlDecode(Text)
    End If
    Return Text
   End Get
  End Property

  Public Shared Function Create(unencodedText As String, contentType As ContentTypes) As BlogMLContent
   Dim content As New BlogMLContent() With {.ContentType = contentType}
   If content.Base64Encoded Then
    Dim byteArray As Byte() = System.Text.Encoding.UTF8.GetBytes(unencodedText)
    content.Text = Convert.ToBase64String(byteArray)
   ElseIf content.HtmlEncoded Then
    content.Text = HttpUtility.HtmlEncode(unencodedText)
   Else
    content.Text = unencodedText
   End If
   Return content
  End Function
 End Class
End Namespace

File: Server/Core/BlogML/Xml/BlogMLContent.cs
using System;
using System.Text;
using System.Web;
using System.Xml.Serialization;

namespace DotNetNuke.Modules.Blog.BlogML.Xml
{
  [Serializable]
  public sealed class BlogMLContent
  {
    private bool Base64Encoded
    {
      get
      {
        return ContentType == ContentTypes.Base64;
      }
    }

    private bool HtmlEncoded
    {
      get
      {
        return ContentType == ContentTypes.Html || ContentType == ContentTypes.Xhtml;
      }
    }

    [XmlAttribute("type")]
    public ContentTypes ContentType
    {
      get
      {
        return m_ContentType;
      }
      set
      {
        m_ContentType = value;
      }
    }
    private ContentTypes m_ContentType = ContentTypes.Text;

    // Encoded Text
    [XmlText]
    public string Text
    {
      get
      {
        return m_Text;
      }
      set
      {
        m_Text = value;
      }
    }
    private string m_Text;

    [XmlIgnore]
    public string UncodedText
    {
      get
      {
        if (Base64Encoded)
        {
          byte[] byteArray = Convert.FromBase64String(Text);
          return Encoding.UTF8.GetString(byteArray);
        }
        if (HtmlEncoded)
        {
          return HttpUtility.HtmlDecode(Text);
        }
        return Text;
      }
    }

    public static BlogMLContent Create(string unencodedText, ContentTypes contentType)
    {
      var content = new BlogMLContent() { ContentType = contentType };
      if (content.Base64Encoded)
      {
        byte[] byteArray = Encoding.UTF8.GetBytes(unencodedText);
        content.Text = Convert.ToBase64String(byteArray);
      }
      else if (content.HtmlEncoded)
      {
        content.Text = HttpUtility.HtmlEncode(unencodedText);
      }
      else
      {
        content.Text = unencodedText;
      }
      return content;
    }
  }
}